quarkus/container/README.md

```diff
@@ -0,0 +1,71 @@
+           IMAGE[:TAG] \
+           start-dev
+
+## Extending the Image
+
+To customize the base image, create a new `Dockerfile` similar to following:
+
+```
+FROM quay.io/keycloak/keycloak
+
+WORKDIR /opt/keycloak
+RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 -dname "CN=server" -alias server -ext "SAN:c=DNS:localhost,IP:127.0.0.1" -keystore conf/server.keystore
+```
+
+In the example above, you are basically creating a new image `FROM` this image and adding a Java Keystore to configure HTTPS using a self-signed certificate (only for example purposes, never do that for production).
+
+As another example, you can install any custom provider you may have (including themes within a JAR file) as follows:
+
+```
+FROM quay.io/keycloak/keycloak
+
+COPY my-providers/ /opt/keycloak/providers/
+
+WORKDIR /opt/keycloak
+
+RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 -dname "CN=server" -alias server -ext "SAN:c=DNS:localhost,IP:127.0.0.1" -keystore conf/server.keystore
+
+# Run the build command to install custom providers
+RUN ./bin/kc.sh build
+```
+
+In the example above, any JAR files within the directory `my-providers` will be copied to the image's `/opt/keycloak/providers` directory. Then the `build` command is executed to install the custom providers.
+
+You can configure the server using any configuration option available in the CLI as follows:
+
+```
+FROM quay.io/keycloak/keycloak
+
+COPY my-providers/ /opt/keycloak/providers/
+
+WORKDIR /opt/keycloak
+
+RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 -dname "CN=server" -alias server -ext "SAN:c=DNS:localhost,IP:127.0.0.1" -keystore conf/server.keystore
+
+RUN ./bin/kc.sh build --db=postgres && ./bin/kc.sh start-dev --db-url=jdbc:postgresql://$DB_HOST/keycloak --db-username=keycloak --db-password=password
+```
+NOTE: Note that we are using separated `RUN` steps for illustrative purposes, ideally you should reduce the number of layers and execute all steps in a single one.
+
+In the example above, Keycloak is set to use e a PostgreSQL database in development mode. You can for sure also use environment variables for the configuration and build an optimized image for better startup time. For more information, see the [Running Keycloak in a container guide](https://www.keycloak.org/guides/server/containers).
+
+
+## Build the image
+
+It is possible to download the Keycloak distribution from a URL:
+
+    docker build --build-arg KEYCLOAK_DIST=http://<HOST>:<PORT>/keycloak-<VERSION>.tar.gz . -t <YOUR_TAG>
+
+Alternatively, you need to build the local distribution first, then copy the distributions tar package in the `containers` folder and point the build command to use the image:
+
+    cp $KEYCLOAK_SOURCE/quarkus/dist/target/keycloak-<VERSION>.tar.gz .
+    docker build --build-arg KEYCLOAK_DIST=keycloak-<VERSION>.tar.gz . -t <YOUR_TAG>
```

Comment:
Was wondering if we should add a pom.xml or a script that does this part?

Reply:
not in this issue, i guess? in general a good idea, but might be evaluated first if we want to do it that way or another? don't know.